crowbar_framework/app/controllers/installer/upgrades_controller.rb

```diff
@@ -153,20 +156,23 @@ def backup
       end
     end
 
-    def nodes
+    def infos
       @current_step = 8
+
       if request.post?
         respond_to do |format|
           begin
-            @service_object.disable_non_core_proposals
-            @service_object.prepare_nodes_for_os_upgrade
+            # TODO(must): Do the following actions async in background
+            # @service_object.disable_non_core_proposals
+            # @service_object.prepare_nodes_for_os_upgrade
+
```

Comment:
> Why, async? disable_non_core_proposals and prepare_nodes_for_os_upgrade won't take too long I think. prepare_nodes_for_os_upgrade in not blocking while performing the upgrade. It will block until the nodes succeed (or failed) calling the reboot command.
> 
> In any case, please leave those steps enabled here even if it's not their final location (and add infos to set_service_object)

This should be async because the following step will do AJAX requests to retrieve the currently still left to update nodes.